### PR TITLE
Implement support for inverting matrices

### PR DESCRIPTION
**Pull request by mkiael**

### CODE REVIEW

File: src/matrix.rs [hunk 1]
```diff
@@ -258,4 +341,141 @@ mod tests {
...
+            [9., 6., 2., 6.],
+            [0., -5., 1., -5.],
+            [0., 0., 0., 0.],
+        ]);
+        assert_eq!(0., a.determinant());
+        assert_eq!(false, a.invertible());
+    }
+
+    #[test]
+    fn inverse_of_matrix() {
```
Comment (johnor) [inline src/matrix.rs:444]: The book has another test for inverse:
"Calculating the inverse of another matrix"

File: src/matrix.rs [hunk 2]
```diff
@@ -85,9 +89,90 @@ macro_rules! define_square_matrix_struct {
...
+            let det = self.determinant();
+            for r in 0..Self::order() {
+                for c in 0..Self::order() {
+                    let cof = self.cofactor(r, c);
+                    res[c][r] = cof / det;
+                }
+            }
+            Ok(res)
+        } else {
+            Err("Matrix is not invertible")
```
Comment (johnor) [inline src/matrix.rs:172]: Is it possible to have a simple test for this case also?

**Review by johnor — APPROVED** [review]
Looks nice! It is good to get compile time errors if the matrix sizes don't match so I like this approach more than the other one :)

File: src/matrix.rs (hunk 1 above)
Comment (mkiael) [reply src/matrix.rs:444]: Missed those, will add!

File: src/matrix.rs (hunk 2 above)
Comment (mkiael) [reply src/matrix.rs:172]: Will fix!

**Pull request merged**